File: src/main/java/com/kmatrokhin/uvbot/entities/UserEntity.java
package com.kmatrokhin.uvbot.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "user", schema = "public")
@Getter
@Setter
public class UserEntity {
    @Id
    private UUID id = UUID.randomUUID();

    private Long chatId;

    private String name;

    private Instant createdAt = Instant.now();
}

File: src/main/java/com/kmatrokhin/uvbot/entities/UserEntity.kt
package com.kmatrokhin.uvbot.entities

import jakarta.persistence.*
import java.time.Instant
import java.util.*

@Entity
@Table(name = "user", schema = "public")
class UserEntity (
    @Id
    var id: UUID = UUID.randomUUID(),
    var chatId: Long,
    var name: String? = null,
    var isSubscribed: Boolean,
    var createdAt: Instant = Instant.now(),

    @Enumerated(EnumType.STRING)
    @Column(columnDefinition = "lang")
    var language: UserLanguage = UserLanguage.ENGLISH
)
